system/jlib/jstring.cpp

```diff
@@ -1595,6 +1595,28 @@ StringBuffer &appendDecodedURL(StringBuffer &s, const char *url)
     return s;
 }
 
+StringBuffer &appendDecodedURL(StringBuffer &s, size_t len, const char *url)
+{
+    size_t i = 0;
```

Comment:
Should this function do something different if it encounters a null char?

Reply:
If anything it should throw an exception - I'm not sure how a url encoded string can include a null.  I think it is probably enough to pass it through - it doesn't validate any other characters are valid.